contracts/drafts/voting/Voting.sol

```diff
@@ -0,0 +1,167 @@
+            "Cannot enact proposal now."
+        );
+        // solium-disable-next-line security/no-call-value
+        (bool success, ) = proposalContract
+            .call.value(msg.value)(proposalData);
+        require(success, "Could not enact proposal.");
+    }
+
+    /**
+     * @dev Use this function to cast votes. Must have approved this contract
+     * (from the frontend) to spend _votes of votingToken tokens.
+     * @param _votes The amount of votingToken tokens that will be casted.
+     */
+    function cast(uint256 _votes) external {
+        require(
+            currentState == "OPEN",
+            "Not open for voting."
+        );
+        IERC20(votingToken).transferFrom(msg.sender, address(this), _votes);
+        if (votes[msg.sender] == 0){
+            voters.push(msg.sender);
+        }
+        votes[msg.sender] = votes[msg.sender].add(_votes);
+        emit VoteCasted(msg.sender, _votes);
+    }
+
+    /**
+     * @dev Use this function to retrieve your votingToken votes in case you changed your mind.
+     */
+    function cancel() external {
+        require(
+            currentState != "SETUP",
+            "Cannot retrieve now."
+        );
+        require(
+            votes[msg.sender] > 0,
+            "No votes casted."
+        );
+        uint256 count = votes[msg.sender];
+        votes[msg.sender] = 0;
+        IERC20(votingToken).transfer(msg.sender, count);
+        emit VoteCancelled(msg.sender, count);
+    }
+
+    /**
+     * @dev Function to open the voting
+     */
+    function open() public onlyOwner {
+        require(
+            quorum > 0,
+            "Quorum not set."
+        );
+        _transition("OPEN");
+    }
+
+    /**
+     * @dev Function to validate the quorum
+     */
+    function validate() public onlyOwner {
+        ERC20Detailed _votingToken = ERC20Detailed(votingToken);
+        int256 totalSupplyFixed = _votingToken.totalSupply()
+            .safeUintToInt().newFixed(_votingToken.decimals());
+        int256 quorumFixed = quorum.safeUintToInt().newFixed(2);
+        uint256 threshold = totalSupplyFixed.multiply(quorumFixed)
+            .fromFixed(_votingToken.decimals()).safeIntToUint();
+        require(
+            _votingToken.balanceOf(address(this)) >= threshold,
+            "Not enough votes to meet the quorum."
+        );
+        _transition("PASSED");
+    }
+
+    /**
+     * @dev Function to cancel all votes
+     */
+    function cancelAllVotes() public onlyOwner {
+        _transition("FAILED");
+    }
+
+    /**
+     * @notice Set the quorum during SETUP state. Quorum must be a percentage of the votingToken's totalSupply, with the comma shifted two digits to the right. For example:
+     * If desired quorum is 100% (unanimity), then pass to this function: 10000.
+     * If desired quorum is 66.67% (super majority), then passs to this function: 6667.
+     */
+    function setQuorum(uint256 _quorum) public onlyOwner {
```

Comment:
Why don't we just pass the quorum / threshold as a parameter to the constructor?